packages/test/functional-tests/src/test/containerRuntime.spec.ts

```diff
@@ -82,41 +117,14 @@ describe("Container Runtime", () => {
 
 		// Function to process an inbound op. It adds delay to simulate time taken in processing an op.
 		function processOp(message: ISequencedDocumentMessage): void {
-			// Simulate batch accumulation by container runtime's remote message processor.
-			// It saves batch messages until the entire batch is received. It calls batch begin
-			// and end on schedule manager accordingly.
-			const batchMetadataFlag = (message.metadata as { batch?: boolean } | undefined)?.batch;
-			let batchStart: boolean = false;
-			let batchComplete: boolean = false;
-			if (batchMetadataFlag === true) {
-				assert(
-					batchClientId === undefined,
-					"Received batch message while another batch is in progress",
-				);
-				batchClientId = message.clientId ?? undefined;
-				batchStart = true;
-			} else if (batchMetadataFlag === false) {
-				assert(batchClientId !== undefined, "Received batch end message without batch");
-				batchClientId = undefined;
-				batchComplete = true;
-			} else if (batchClientId === undefined) {
-				batchStart = true;
-				batchComplete = true;
-			}
-
-			if (batchStart) {
-				scheduleManager.batchBegin(message);
-			}
-
-			// Add delay such that each op takes greater than the DeltaScheduler's processing time to process.
+			// Add delay to container runtime's op processing such that each op takes greater than the
+			// DeltaScheduler's processing time to process.
 			const processingDelay = DeltaScheduler.processingTime + 10;
-			const startTime = Date.now();
-			while (Date.now() - startTime < processingDelay) {}
-
-			if (batchComplete) {
-				scheduleManager.batchEnd(undefined /* error */, message);
-			}
-
+			containerRuntime.once("op", () => {
+				const startTime = Date.now();
+				while (Date.now() - startTime < processingDelay) {}
```

Comment:
Side note -- Is this hard on the CPU? Better to go async here...?

Reply:
I changed this to use sinon timers.